django/db/models/fields/related.py

```diff
@@ -1662,12 +1662,7 @@ def value_from_object(self, obj):
         """
         Return the value of this field in the given model instance.
         """
-        if obj.pk is None:
```

Comment:
I think we'll need to keep the `if obj.pk is None` condition to fix the test failure.